engine/src/main/java/org/terasology/logic/characters/CharacterSystem.java

```diff
@@ -73,10 +74,21 @@
     @In
     private Time time;
 
+    @In
+    private TranslationSystem translationSystem;
+
     @ReceiveEvent(components = {CharacterComponent.class})
     public void onDeath(DoDestroyEvent event, EntityRef entity) {
         CharacterComponent character = entity.getComponent(CharacterComponent.class);
-        character.controller.send(new DeathEvent());
+        DeathEvent deathEvent = new DeathEvent();
+        //Store the details of the death in the event for display on the death screen
+        deathEvent.lastDamageType = event.getDamageTypeString();
+        if (event.getInstigatorString() != null) {
+            deathEvent.lastInstigator = event.getInstigatorString();
+        } else {
```

Comment:
I think this class is a better place to convert the prefabs/entities to readable names, i'd expect little to none logic in events but in systems instead. Move the two conversion methods from the event to this system.